src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -938,7 +959,6 @@ private function addLockSection(ArrayNodeDefinition $rootNode)
                         })
                     ->end()
                     ->addDefaultsIfNotSet()
-                    ->fixXmlConfig('resource')
```

Comment:
won't this prevent using `<resource>` tags in xml configuration files?

Reply:
@nicolas-grekas As I mentioned in the issue:

The reason for this is cause `->fixXmlConfig('resource')` happenes after all the `beforeNormalization` rules, one of which is:

```php
    ->beforeNormalization()
        ->ifTrue(function ($v) { return \is_array($v) && !isset($v['resources']); })
        ->then(function ($v) {
            $e = $v['enabled'];
            unset($v['enabled']);

            return ['enabled' => $e, 'resources' => $v];
        })
    ->end()
```

so you get:

```php
    'resources' => [
        'resource' => [
            [
                'name'  => 'foo',
                'value' => 'semaphore',
            ],
            [
                'name'  => 'bar',
                'value' => 'flock',
            ],
        ],
    ],
```

Since I added a new `beforeNormalization` rule, `fixXmlConfig` is no longer required.